docs/adr/0037-use-nx-for-monorepo.md

```diff
@@ -0,0 +1,48 @@
+# Use nx for monorepo tooling
+
+- Status: proposed
+- Deciders: Ben Bangert, Wil Clouser, Dan Schomburg
+- Date: 2023-03-13
+- Obsoletes: 0019-use-workspace-dependencies
+
+## Context and Problem Statement
+
+Our monorepo using a yarn workspace model has a few limitations we've experienced such as slow builds, using bash scripts for running tasks, having to make each new library or component as its own package with its own dependencies, and an inflexible structure for refactoring our shared libraries.
+
+Monorepo tooling has progressed since the prior ADR on using workspace dependencies with yarn and we'd like to take advantage of these improvements.
+
+## Decision Drivers
+
+- Faster builds with dependency structure analysis and build caching.
+- Easy to migrate into without large changes to our existing setup.
+- Tooling built for monorepo needs.
+- Task running, caching, and dependency handling
+- Easy to create/share/move library/app code for re-use and encapsulation.
+
+## Considered Options
+
+- A. nx
+- B. Turborepo
+
+## Decision Outcome
+
+Chosen option: A. nx. Both options provide powerful build capabilities with similar caching options. nx is a little ahead on additional monorepo tooling that would benefit some of our upcoming work. nx supports two models of monorepo, a package-based approach (what FxA current is built with), and an integrated approach that utilizes TypeScript paths for easily splitting out and re-organizing code. This allows it to work almost as-is with our existing monorepo. We can then integrate/refactor existing code into lighter-weight integrated-style components for use in our applications.
```

Comment:
> ... an inflexible structure for refactoring our shared libraries

> We can then integrate/refactor existing code into lighter-weight integrated-style components for use in our applications.

Not necessary but it could be good to give one small example of what this looks like, e.g. moving something out of our too-large `fxa-shared` directory and do what with it to make it a "lighter-weight integrated-style component"?

Reply:
Rewrote to highlight basic directory structure approach that the integrated approach results in.